src/components/signup/SignUp.js

```diff
@@ -0,0 +1,157 @@
+import "./SignUp.css";
+import { FaTwitter } from "react-icons/fa";
+import { useState } from "react";
+import { axiosInstanceReg } from "../api/axios";
+import Login from "../login/Login";
+
+const SignUp = () => {
+  const [details, setDetails] = useState({
+    first_name: "",
+    last_name: "",
+    username: "",
+    email: "",
+    password: "",
+    password_confirmation: "",
+  });
+  const [errMsg, setErrMsg] = useState("");
+  const [success, setSuccess] = useState(false);
+
+  const handleChange = (e) => {
+    setDetails((state) => ({
+      ...state,
+      [e.target.name]: e.target.value.trim(),
+    }));
+  };
+
+  const postRequest = async () => {
+    try {
+      await axiosInstanceReg.post("/register", details);
+      setSuccess(true);
+    } catch (err) {
+      console.log(err);
+      setErrMsg(err.message);
+      setSuccess(false);
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+      postRequest()
+  };
+
+
+  return (
+    <>
+      {success ? (
+        <Login />
+      ) : (
+        <div className="signup-conatiner">
+          <div className="signup-split signup-left">
+            <div className="signup-centered">
+              <h1>Twitter</h1>
+              <FaTwitter className="singup-logo" />
+            </div>
+          </div>
+
+          <div className="signup-split signup-right">
+            <h2 className="signup-form-header">Signup</h2>
+            <form className="signup-centered" onSubmit={handleSubmit}>
+              <div className="signup-err">
+                <span className="signup-error">{errMsg}</span>
+              </div>
```

Comment:
Should this not be conditionally rendered?

Reply:
I will adjust this